src/physics/euler.cpp

```diff
@@ -569,6 +611,77 @@ ::convective_surface_numerical_split_flux (
     return flux_interp_to_surface;
 }
 
+template <int dim, int nstate, typename real>
+std::array<real,nstate> Euler<dim, nstate, real>
+::compute_entropy_variables (
+    const std::array<real,nstate> &conservative_soln) const
+{
+    std::array<real,nstate> entropy_var;
+    const real density = conservative_soln[0];
+    const real pressure = compute_pressure<real>(conservative_soln);
+    const dealii::Tensor<1,dim,real> vel = compute_velocities<real>(conservative_soln);
+    const real vel2 = compute_velocity_squared<real>(vel);
+    const real entropy = log(pressure) - gam * log(density);
+    const real rho_theta = conservative_soln[nstate-1] - 0.5 * density * vel2;
```

Comment:
for efficiency, you could equivalently write `rho_theta = pressure/gamm1` instead if you're not using `vel` and `vel2`